src/net.cpp

```diff
@@ -2358,22 +2236,36 @@ bool CConnman::BindListenPort(const CService& addrBind, bilingual_str& strError,
 
     // Allow binding if the port is still in TIME_WAIT state after
     // the program was closed and restarted.
-    setsockopt(sock->Get(), SOL_SOCKET, SO_REUSEADDR, (sockopt_arg_type)&nOne, sizeof(int));
+    if (sock->SetSockOpt(SOL_SOCKET, SO_REUSEADDR, (sockopt_arg_type)&nOne, sizeof(int)) == SOCKET_ERROR) {
```

Comment:
please replace this c-style cast with a reinterpret_cast (I think)

Reply:
See https://github.com/bitcoin/bitcoin/pull/21878#discussion_r804795139